### PR TITLE
fix(learn): add hint to implement-map-and-filter-on-prototype

### LINKED ISSUE #39486: Update challenges with hint on how to get to the array
Challenges: 
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/implement-the-filter-method-on-a-prototype

The assumption that the camper will know to use `this` inside the callback to get to the array is pretty unfounded in my opinion. The challenge text gives no hints or information about this what so ever.

I think this challenge should be updated to include some sort of hint, or just straight up tell the camper how to get to the array inside the callback.

### PR DESCRIPTION
**Pull request by nibble0101**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39486

This pull request updates the following challenges by giving a hint to the learner on how to access the `Array` instance while implementing custom `map` and `filter`  array methods.
-  [implement map on a prototype](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype)
- [implement the filter method on a prototype](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/implement-the-filter-method-on-a-prototype)

### CODE REVIEW

**Comment by RandellDawson** [comment]
@nibble0101 If we accept the additional note, it would be best to use markdown syntax instead of `ul` and `li` elements.  You can create a list by prefixing the line with `- ` or `* `.

**Comment by nibble0101** [comment]
@RandellDawson  I did use markdown at first but after checking on Gitpod, the list isn't displaying.

**Comment by RandellDawson** [comment]
You must have a empty single line after the the opening `section` tag and before the closing `section` tag for the markdown to work.  The documentation does not reflect this currently, but since we will soon be using a completely different challenge file template, there is no need to update the documentation at this time.

**Review by raisedadead — CHANGES_REQUESTED** [review]
The `pathMigration` file is no longer needed. Can you rebase against master to get the latest changes, thanks.

**Comment by nibble0101** [comment]
@raisedadead  
> Can you rebase against master to get the latest changes.

What does that mean?

**Comment by raisedadead** [comment]
> What does that mean?

https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=conflicts-on-a-pull-request

**Comment by RandellDawson** [comment]
> What does that mean?

@nibble0101 

```sh
git pull --rebase upstream master
```
This will get all the changes that might have been merged to `master` since you first started working on this PR and then adds your commits on top.  If there are any merge conflicts, you will have to resolve them.

Also, you want to make sure to remove the `api-server/server/resources/pathMigration.json` which got created.

**Comment by nibble0101** [comment]
@RandellDawson  Thanks. I keep getting "Current branch is up to date" whether i run that command from the main branch or from the i am working on.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype.english.md [hunk 1]
```diff
@@ -6,19 +6,27 @@ forumTopicId: 301230
 ---
 
 ## Description
+
 <section id='description'>
 As you have seen from applying <code>Array.prototype.map()</code>, or simply <code>map()</code> earlier, the <code>map</code> method returns an array of the same length as the one it was called on. It also doesn't alter the original array, as long as its callback function doesn't.
 In other words, <code>map</code> is a pure function, and its output depends solely on its inputs. Plus, it takes another function as its argument.
 It would teach us a lot about <code>map</code> to try to implement a version of it that behaves exactly like the <code>Array.prototype.map()</code> with a <code>for</code> loop or <code>Array.prototype.forEach()</code>.
-Note: A pure function is allowed to alter local variables defined within its scope, although, it's preferable to avoid that as well.
+Note:
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/implement-map-on-a-prototype.english.md:14]:
```suggestion
**Notes:** 
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/implement-the-filter-method-on-a-prototype.english.md [hunk 2]
```diff
@@ -6,17 +6,24 @@ forumTopicId: 301231
 ---
 
 ## Description
+
 <section id='description'>
 It would teach us a lot about the <code>filter</code> method if we try to implement a version of it that behaves exactly like <code>Array.prototype.filter()</code>. It can use either a <code>for</code> loop or <code>Array.prototype.forEach()</code>.
-Note: A pure function is allowed to alter local variables defined within its scope, although, it's preferable to avoid that as well.
+Note:
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/implement-the-filter-method-on-a-prototype.english.md:12]:
```suggestion
**Notes:** 
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
I am not opposed to adding the extra note as we are not planning on completely rewriting or rearranging the other challenges that would cover `this` for thoroughly.

I just have two small formatting changes.

**Comment by moT01** [comment]
I'm not a fan of how big the text gets when there's a list...

Screen Shot 2020-09-14 at 11 14 08 AM

It makes the notes feel more important than the description. I'm really not a fan of the list itself either. I feel like these notes could be integrated into the description and instructions. Maybe something like this...?

Screen Shot 2020-09-14 at 11 33 37 AM

After trying to come up with a suggestion, I felt there was a lot that could be changed up. I didn't see a need for the first note. Perhaps I am getting outside the scope of this issue, but I feel like it should look something like the above. If we don't want to rewrite it that much, I still vote for integrating the notes into the description/instructions. That my opinion, although, if others don't want to go that route, I can live with it.

**Comment by RandellDawson** [comment]
@moT01 Inserting the notes into the instructions section is fine too.  However, for the last sentence of the description section, I would prefer:

> You could use a `for` loop or the `Array.prototype.forEach` method.

The challenges should be written from the second person point of view.

**Comment by nibble0101** [comment]
@RandellDawson  and @moT01 . What is the final decision here?

**Comment by moT01** [comment]
I think we want to go with my suggestion and Randell's modification - I dunno, what do you think of that @nibble0101?

**Comment by RandellDawson** [comment]
@nibble0101 I discussed with @moT01 that I would be on board with using his suggestion with my adjustment.

**Comment by nibble0101** [comment]
@RandellDawson, @moT01. I am okay with your suggestions above.

**Comment by nibble0101** [comment]
@RandellDawson  @moT01  @raisedadead  I am completely confused by these git commands. I am not sure I am doing the right thing anymore. I closed this PR without knowing twice already. I am closing this PR and  and opening another one.

**Pull request closed without merging**